01-data-structures/02-stacks-and-queues/myqueue/myqueue.rb

```diff
@@ -8,11 +8,38 @@ def initialize
   end
 
   def enqueue(element)
+    if @queue.empty?
+      @queue[0] = element
+    else
+      new_array = [element]
+      @queue.each_with_index do |person, index|
+        new_array[index + 1] = @queue[index]
+      end
+      @queue = new_array
+    end
+    @head = @queue[0]
+    @tail = @queue[-1]
   end
 
   def dequeue
+    if @queue.empty?
+      return true
+    else
+      second_array = []
+      @queue.each_with_index do |person, index|
+        @queue[index - 1]
```

Comment:
While I am reasonably confident this actually works, I am not sure it is doing what you might think it is doing. I would be interested to hear your rationale if you think you do understand what exactly happens here.